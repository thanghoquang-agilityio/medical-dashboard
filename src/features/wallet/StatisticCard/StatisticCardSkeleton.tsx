import { Button } from '@/components/ui';
import { MoreIcon } from '@/icons';
import { Card, CardBody, CardHeader, Skeleton } from '@nextui-org/react';
import React from 'react';

export type StatisticCardSkeletonProps = {
  title: string;
};
export const StatisticCardSkeleton = ({
  title,
}: StatisticCardSkeletonProps) => {
  return (
    <Card
      as="section"
      className="bg-background-200 min-w-[236px] w-full py-[14px] h-[140px] px-5 flex flex-col gap-2"
    >
      <CardHeader className="p-0 flex justify-between">
        <h3 className="text-[14px]/8 font-bold text-primary-200">{title}</h3>
        <Button
          isIconOnly
          className="min-w-[11px] w-[11px] h-[11px] md:w-4 md:h-4"
        >
          <MoreIcon customClass="w-[11px] h-[11px] md:w-4 md:h-4 rotate-90" />
        </Button>
      </CardHeader>
      <CardBody className="p-0">
        <Skeleton className="h-4 rounded-medium" />
      </CardBody>
    </Card>
  );
};
