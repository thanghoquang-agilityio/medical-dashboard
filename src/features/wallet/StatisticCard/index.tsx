import { Button, Text } from '@/components/ui';
import { MoreIcon } from '@/icons';
import { formatNumberWithUnit } from '@/utils';
import { Card, CardBody, CardHeader } from '@nextui-org/react';
import React from 'react';

export type StatisticCardProps = {
  title: string;
  value: number;
};
const StatisticCard = ({ title, value }: StatisticCardProps) => {
  return (
    <Card
      as="section"
      className="bg-background-200 min-w-[236px] w-full py-[14px] h-[140px] px-5 flex flex-col gap-2 font-manrope"
    >
      <CardHeader className="p-0 flex justify-between">
        <h3 className="text-[14px]/8 font-bold text-primary-200">{title}</h3>
        <Button
          isIconOnly
          className="min-w-[11px] w-[11px] h-[11px] md:w-4 md:h-4"
        >
          <MoreIcon customClass="w-[11px] h-[11px] md:w-4 md:h-4 rotate-90" />
        </Button>
      </CardHeader>
      <CardBody className="p-0">
        <Text size="2xl" variant="primary" customClass="font-bold leading-8">
          ${formatNumberWithUnit(value)}
        </Text>
      </CardBody>
    </Card>
  );
};

StatisticCard.displayName = 'StatisticCard';
export default StatisticCard;
