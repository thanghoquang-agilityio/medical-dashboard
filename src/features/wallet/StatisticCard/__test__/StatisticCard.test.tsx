import { render } from '@testing-library/react';
import StatisticCard, { StatisticCardProps } from '..';

describe('StatisticCard', () => {
  const setup = (props: StatisticCardProps) =>
    render(<StatisticCard {...props} />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  const mockProps: StatisticCardProps = {
    title: 'Mock Title',
    value: 100,
  };

  it('should render correctly', () => {
    const { asFragment } = setup(mockProps);

    expect(asFragment()).toMatchSnapshot();
  });
});
