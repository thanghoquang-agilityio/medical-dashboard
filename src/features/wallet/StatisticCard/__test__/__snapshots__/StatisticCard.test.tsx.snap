// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticCard should render correctly 1`] = `
<DocumentFragment>
  <section
    class="relative overflow-hidden text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none bg-background-200 min-w-[236px] w-full py-[14px] h-[140px] px-5 flex flex-col gap-2"
    tabindex="-1"
  >
    <div
      class="z-10 w-full items-center shrink-0 overflow-inherit color-inherit subpixel-antialiased rounded-t-large p-0 flex justify-between"
    >
      <h3
        class="text-[14px]/8 font-bold text-primary-200"
      >
        Mock Title
      </h3>
      <button
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small gap-2 rounded-medium px-0 !gap-0 transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover border-transparent font-semibold bg-linear-success text-content1 bg-none min-w-[11px] w-[11px] h-[11px] md:w-4 md:h-4"
        type="button"
      >
        <svg
          class="w-[11px] h-[11px] md:w-4 md:h-4 rotate-90"
          fill="none"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 6C6.90076 6 6 6.90076 6 8C6 9.09924 6.90076 10 8 10C9.09924 10 10 9.09924 10 8C10 6.90076 9.09924 6 8 6Z"
            fill="currentColor"
          />
          <path
            d="M8 0C6.90076 0 6 0.900764 6 2C6 3.09924 6.90076 4 8 4C9.09924 4 10 3.09924 10 2C10 0.900764 9.09924 0 8 0Z"
            fill="currentColor"
          />
          <path
            d="M8 12C6.90076 12 6 12.9008 6 14C6 15.0992 6.90076 16 8 16C9.09924 16 10 15.0992 10 14C10 12.9008 9.09924 12 8 12Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <div
      class="relative flex w-full flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased p-0"
    >
      <p
        class="text-primary-100 text-2xl font-bold leading-8"
      >
        $100 
      </p>
    </div>
  </section>
</DocumentFragment>
`;
